packages/protocol/test-sol/unit/governance/validators/Validators.t.sol

```diff
@@ -220,14 +213,22 @@ contract ValidatorsTest is Test, TestConstants, Utils, ECDSAHelper {
 
   function _whenL2WithEpoch() public {
     uint256 l1EpochNumber = IPrecompiles(address(validators)).getEpochNumber();
-    deployCodeTo("Registry.sol", abi.encode(false), PROXY_ADMIN_ADDRESS);
+    _whenL2();
 
     address[] memory _elected = new address[](2);
     _elected[0] = validator;
     _elected[1] = otherValidator;
     epochManager.initializeSystem(l1EpochNumber, block.number, _elected);
   }
 
+  function _getEpochNumberBasedOnLayer() internal returns (uint256) {
+    if (isL2()) {
+      return epochManager.getCurrentEpochNumber();
+    } else {
+      return IPrecompiles(address(validators)).getEpochNumber();
+    }
+  }
+
```

Comment:
Shouldn't this be moved to some parent class?

Reply:
Actually, this logic is already implemented in PrecompilesOverride. Created a V2 version (with hardcoded registry, to avoid `owner` variable clash) and use that in this test.